.github/workflows/e2e.yaml

```diff
@@ -5,6 +5,11 @@ permissions: read-all
 jobs:
   test:
     runs-on: ubuntu-latest
+    container:
+      image: golang:1.20-bullseye
```

Comment:
Question, do we need to run in Go docker image? It undermines the work we put in to have single place that specifies Go version.  I would assume that block below should install a proper Go version.
```
      - id: goversion
        run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
      - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
        with:
          go-version: ${{ steps.goversion.outputs.goversion }}
```

Reply:
It's a fair question, we don't need to.  I think originally I chose the golang based image so that if necessary I could "docker exec" into the container to test things and verify steps the ci would run were going to work as expected.

It gives us a way to test things without having to replicate what `actions/setup-go`  would do, and also doesn't impact/hinder the version of go installed by `actions/setup-go` so I think my preference would be to maintain current image rather than step back to pure debian or ubuntu image without golang present.

Reply:
I expect that `actions/setup-go`  installs go version from binaries that is independent to one from `golang:1.20-bullseye`. Might be wrong though.